File: queue/queue.js
var Queue = (function () {
    function Queue(size) {
        this.first = -1;
        this.last = -1;
        this.numOfElements = 0;
        this.queue = new Array(size);
        this.queueSize = size;
    }
    ;
    Queue.prototype.enqueue = function (ele) {
        if (this.last + 1 > this.queueSize) {
            return;
        }
        this.queue[this.last] = ele;
        this.last = this.last + 1;
        this.numOfElements = this.numOfElements + 1;
    };
    ;
    Queue.prototype.dequeue = function () {
        if (this.numOfElements <= 0) {
            return -1;
        }
        var firstEle = this.queue[this.first];
        this.first = this.first + 1;
        this.numOfElements = this.numOfElements - 1;
        return firstEle;
    };
    ;
    return Queue;
}());
// main
(function () {
    var q = new Queue(10);
    q.enqueue(1);
    q.enqueue(7);
    console.log(q.dequeue());
})();

File: queue/Queue.java
package queue;

public class Queue {
    private int queue[];
    private int queueSize;

    private int firstIndex = 0;
    private int lastIndex = -1;
    private int numOfElements = 0;

    public Queue(int queueSize) {
        this.queue = new int[queueSize];
        this.queueSize = queueSize;
    }

    public void enqueue(int num) {
        if (this.lastIndex + 1 >= this.queueSize) {
            return;
        }

        this.queue[this.lastIndex + 1] = num;
        this.lastIndex++;
        this.numOfElements++;
    }

    public int dequeue() {
        if (this.numOfElements <= 0) {
            return -1;
        }

        int firstNum = this.queue[this.firstIndex];
        this.firstIndex++;
        this.numOfElements--;
        return firstNum;
    }

    public static void main(String[] args) {
        Queue q = new Queue(5);
        q.enqueue(5);
        q.enqueue(4);
        q.enqueue(3);
        q.enqueue(2);
        q.enqueue(1);
        System.out.println(q.dequeue());
        System.out.println(q.dequeue());
        System.out.println(q.dequeue());
        System.out.println(q.dequeue());
        System.out.println(q.dequeue());
    }
}
